fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/AccountNumberGenerator.java

```diff
@@ -154,6 +182,36 @@ private String generateAccountNumber(Map<String, String> propertyMap, AccountNum
         return accountNumber;
     }
 
+    private String randomNumberGenerator(int accountMaxLength, Map<String, String> propertyMap) {
+        String randomNumber = RandomStringUtils.random(accountMaxLength, false, true);
+        String accountNumber = randomNumber.substring(0, accountMaxLength);
+        BigInteger accNumber = new BigInteger(accountNumber);
+
+        if (!accNumber.equals(BigInteger.ZERO)) { // to avoid account no. 00 in randomisation
+            String entityType = propertyMap.get(ENTITY_TYPE);
+            if (entityType.equals("client")) { // avoid duplication it will loop until it finds new random account no.
+
+                Client client = this.clientRepository.getClientByAccountNumber(accountNumber);
```

Comment:
What if the account numbers have prefixes / shortnames added to them? Then this will never return a match even though there may be matching ones after we append the prefix...